### PR TITLE
Use identity comparison with singleton

### PR DESCRIPTION
**Pull request by ThibFrgsGmz**
| | |
|:---|:---|
|**_Originating Project/Creator_**| @ThibFrgsGmz |
|**_Affected Component_**| python autocoder |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**| void |
|**_Has Unit Tests (y/n)_**| n |
|**_Builds Without Errors (y/n)_**| Let CI run |
|**_Unit Tests Pass (y/n)_**| Let CI run |
|**_Documentation Included (y/n)_**| n |

---
## Change Description

This PR simply changes the way we compare a variable with the singleton objects ``True``, ``False``, and ``None``.
 
## Rationale

Comparisons with singleton objects such as ``True``, ``False``, and ``None`` should be made with identity rather than equality (either ``is`` or ``is not``). [ref](https://stackoverflow.com/questions/2209755/python-operation-vs-is-not)

Identity checks are more efficient than equality checks. In some cases, equality checks can also result in unintended behavior.

## Testing/Review Recommendations

Sorry for the code review which requires a certain mental load. Since the complexity of the changes is very low I didn't think it was worth splitting it.

## Future Work

 void

### CODE REVIEW

**Review by LeStarch — APPROVED** [review]
Looks good, but shouldn't we just check true or false rather than `is True` and `is False`?

**Comment by ThibFrgsGmz** [comment]
You are right, especially since [PEP8](https://pep8.org/) also recommends it. This is done now.

File: Autocoders/Python/src/fprime_ac/utils/XmlParser.py [hunk 1]
```diff
@@ -207,7 +207,7 @@ def parse_file(self, filename):
                 # requirement for parsing a second file is that it has a same
                 # named root Element.
 
-                if self.__root is not None and root_name_checked == False:
+                if self.__root is not None and root_name_checked:
```
Comment (LeStarch) [inline Autocoders/Python/src/fprime_ac/utils/XmlParser.py:210]: This changed logic, I think.  Seems it should read:

```
if self.__root is not None and not root_name_checked:
```

File: Autocoders/Python/src/fprime_ac/utils/ParseC.py [hunk 2]
```diff
@@ -216,7 +216,7 @@ def ParseTypedefEnum(typename, filename, loadfile=True):
     #
     # LJR Added 30 July 2007 to flag missing enum definitions.
     #
-    if (typename in data) == False:
+    if (typename in data):
```
Comment (LeStarch) [inline Autocoders/Python/src/fprime_ac/utils/ParseC.py:219]: Same here:

```
if typename not in data:
```

File: Autocoders/Python/src/fprime_ac/generators/formatters.py [hunk 3]
```diff
@@ -1339,7 +1339,7 @@ def subThreadInstanceFirstArg(self, mod_id, args, context_list):
         # if context_list is None:
         #    return args
 
-        if self.subThreadTest(mod_id) == True and len(context_list) > 0:
+        if self.subThreadTest(mod_id) is True and len(context_list) > 0:
```
Comment (LeStarch) [inline Autocoders/Python/src/fprime_ac/generators/formatters.py:1342]: This one should be:

```
if self.subThreadTest(mod_id) and len(context_list) > 0:
```

File: Autocoders/Python/src/fprime_ac/generators/formatters.py [hunk 4]
```diff
@@ -1010,7 +1010,7 @@ def formatFunComment(self, name, args, proto=True, ident=0):
         new_list = []
         for line in type_args_list[:-1]:
             new_list.append(line + ",")
-        if proto == True:
+        if proto is True:
```
Comment (LeStarch) [inline Autocoders/Python/src/fprime_ac/generators/formatters.py]:
```
if proto:
```

**Review by LeStarch — CHANGES_REQUESTED** [review]
I found a few nits where I think logic changed.  I'd have fixed it, but I want another opinion, so forgive the pass-back.

File: Autocoders/Python/src/fprime_ac/utils/ParseC.py (hunk 2 above)
Comment (ThibFrgsGmz) [reply Autocoders/Python/src/fprime_ac/utils/ParseC.py:219]: Yes

File: Autocoders/Python/src/fprime_ac/generators/formatters.py (hunk 3 above)
Comment (ThibFrgsGmz) [reply Autocoders/Python/src/fprime_ac/generators/formatters.py:1342]: Yes

File: Autocoders/Python/src/fprime_ac/generators/formatters.py (hunk 4 above)
Comment (ThibFrgsGmz) [reply Autocoders/Python/src/fprime_ac/generators/formatters.py]: And yes

File: Autocoders/Python/src/fprime_ac/utils/XmlParser.py (hunk 1 above)
Comment (ThibFrgsGmz) [reply Autocoders/Python/src/fprime_ac/utils/XmlParser.py]: Fixed.

File: Autocoders/Python/src/fprime_ac/generators/formatters.py (hunk 3 above)
Comment (ThibFrgsGmz) [reply Autocoders/Python/src/fprime_ac/generators/formatters.py]: Fixed.

File: Autocoders/Python/src/fprime_ac/generators/formatters.py (hunk 4 above)
Comment (ThibFrgsGmz) [reply Autocoders/Python/src/fprime_ac/generators/formatters.py]: Fixed.

File: Autocoders/Python/src/fprime_ac/utils/ParseC.py (hunk 2 above)
Comment (ThibFrgsGmz) [reply Autocoders/Python/src/fprime_ac/utils/ParseC.py]: Fixed.

**Review by LeStarch — APPROVED** [review]

**Pull request merged**